tests/ceph_ansible/test_ansible_upgrade.py

```diff
@@ -182,11 +182,25 @@ def compare_ceph_versions(pre_upgrade_versions, post_upgrade_versions):
     for name, version in pre_upgrade_versions.items():
         # skipping node-exporter version check as it not supported
         if name != "node-exporter":
-            if "installer" not in name and post_upgrade_versions[name] == version:
-                log.error("Pre upgrade version matches post upgrade version")
-                log.error("{}: {} matches".format(name, version))
-                return 1
-    return 0
+            try:
+                if "installer" not in name and post_upgrade_versions[name] == version:
+                    log.error("Pre upgrade version matches post upgrade version")
+                    log.error("{}: {} matches".format(name, version))
+                    return 1
+            except KeyError:
+                # for handling rgw conatiner names during 3.x 'some-rgw' but in 4.x 'some-rgw-rgw0'
```

Comment:
I believe it would be a good idea to check the build version. We can ignore if it is an upgrade from 3.x to 4.x...

Reply:
Even with 3.x to 4.x container upgrade we can't ignore version check tests as some of the containers may not be upgraded
hence handled it with Exceptions.

Reply:
I swallowed some words... what I meant was as the code is generic... we can ignore if it is an upgrade `other than` from 3.x to 4.x

Reply:
The reason for the statement is, we could do something like
```
if pre_upgrade_version.startswith('3') and 'rgw' in name:
  do_something_different
else:
  check if the versions are same.
```

Reply:
It is easier to under what is happening. Without your comment, it would be difficult why the exception is handled also, we iterate through the list of keys everytime there is an exception.

Reply:
added `break` so it won't iterate over all keys also It won't hit exception every time other than rgw node name
>>if pre_upgrade_version.startswith('3') and 'rgw' in name:
  do_something_different
  
sorry, but still I am not sure what to to add here `do_something_different`
Can you please let me know your logic?

Reply:
Thanks @psathyan for your suggestion and logic for optimizing code and readability.

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1616670284897/result.html